apps/election-manager/src/components/ExportElectionBallotPackageModalButton.tsx

```diff
@@ -0,0 +1,321 @@
+  }
+
+  const now = new Date()
+  const defaultFileName = generateFilenameForBallotExportPackage(
+    electionDefinition!,
+    now
+  )
+
+  // Callback to open the file dialog.
+  const openFileDialog = async () => {
+    if (state.type !== 'ArchiveBegin') {
+      throw new Error(
+        `unexpected state '${state.type}' found during openFileDialog callback`
+      )
+    }
+    try {
+      await state.archive.begin({
+        defaultPath: defaultFileName,
+        filters: [{ name: 'Archive Files', extensions: ['zip'] }],
+      })
+      await state.archive.file('election.json', electionData)
+      await state.archive.file(
+        'manifest.json',
+        JSON.stringify({ ballots: state.ballotConfigs }, undefined, 2)
+      )
+
+      setState(workflow.next)
+    } catch (error) {
+      setState(workflow.error(state, error))
+    }
+  }
+
+  let mainContent = null
+  let actions = null
+
+  switch (state.type) {
+    case 'Init': {
+      mainContent = <Loading />
+      actions = (
+        <LinkButton onPress={closeModal} disabled>
+          Cancel
+        </LinkButton>
+      )
+      break
+    }
+
+    case 'ArchiveBegin':
+      switch (usbDriveStatus) {
+        case UsbDriveStatus.absent:
+        case UsbDriveStatus.notavailable:
+        case UsbDriveStatus.recentlyEjected:
+          actions = <LinkButton onPress={closeModal}>Cancel</LinkButton>
+          mainContent = (
+            <Prose>
+              <h1>No USB Drive Detected</h1>
+              <p>
+                <USBImage src="usb-drive.svg" alt="Insert USB Image" />
+                Please insert a USB drive in order to export the ballot
+                configuration.
+              </p>
+            </Prose>
+          )
+          break
+        case UsbDriveStatus.ejecting:
+        case UsbDriveStatus.present:
+          mainContent = <Loading />
+          actions = (
+            <LinkButton onPress={closeModal} disabled>
+              Cancel
+            </LinkButton>
+          )
+          break
+        case UsbDriveStatus.mounted: {
+          // TODO(caro): Update this to just write the file to the USB drive once the APIS in kiosk-browser exist.
+          actions = (
+            <React.Fragment>
+              <LinkButton onPress={closeModal}>Cancel</LinkButton>
+              <Button onPress={openFileDialog}>Custom</Button>
+              <Button onPress={openFileDialog} primary>
+                Export
+              </Button>
+            </React.Fragment>
+          )
+          mainContent = (
+            <Prose>
+              <h1>Export Ballot Package</h1>
+              <p>
+                <USBImage src="usb-drive.svg" alt="Insert USB Image" />
+                Would you like to export the ballot configuration now? A zip
+                archive will automatically be saved to the default location on
+                the mounted USB drive. Optionally, you may pick a custom export
+                location.
```

Comment:
First sentence doesn't seem necessary. Let's remove it to have less text on this modal.
```suggestion
                A zip archive will automatically be saved to the default location on
                the mounted USB drive. Optionally, you may pick a custom export
                location.
```